### PR TITLE
Update Javadoc of premium SDK for azure-resourcemanager-resources

### PR DESCRIPTION
**Pull request by v-hongli1**
- https://github.com/Azure/azure-sdk-for-java/issues/36596

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/dag/TaskItem.java [hunk 1]
```diff
@@ -14,6 +14,8 @@
  */
 public interface TaskItem {
     /**
+     * Gets the result of the task invocation
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/dag/TaskItem.java:17]: nit, `.` at the end.

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/model/HasInnerModel.java [hunk 2]
```diff
@@ -10,6 +10,8 @@
  */
 public interface HasInnerModel<T> {
     /**
+     * Gets
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/model/HasInnerModel.java:13]: incomplete comment

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/models/Deployment.java [hunk 3]
```diff
@@ -34,67 +34,93 @@ public interface Deployment extends Indexable, Refreshable<Deployment>, Updatabl
     HasInnerModel<DeploymentExtendedInner>, HasManager<ResourceManager>, HasName, HasId {
 
     /**
+     * Gets the name of this deployment's resource group
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/models/Deployment.java:37]: same, `.` at the end. check other places as well

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/dag/DAGraph.java [hunk 4]
```diff
@@ -76,6 +80,8 @@ public boolean isRootNode(NodeT node) {
     }
 
     /**
+     * this dag is the preparer responsible for preparing the DAG for traversal.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/dag/DAGraph.java:83]:
```suggestion
     * Check whether this dag is the preparer responsible for preparing the DAG for traversal.
```

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
From CI, you have more to solve.

**Comment by v-hongli1** [comment]
> From CI, you have more to solve.

Indeed, there are more and more Javadoc that need to be corrected.

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/dag/Node.java [hunk 5]
```diff
@@ -48,41 +48,53 @@ public Node(final String key, final DataT data) {
...
+     * Gets children (neighbours) of this node.
+     *
      * @return children (neighbours) of this node
      */
     public List<String> children() {
         return Collections.unmodifiableList(this.children);
     }
 
     /**
+     * Sets child (neighbour) to this node.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/dag/Node.java:87]: nit, "Adds ..

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java [hunk 6]
```diff
@@ -26,7 +26,13 @@
...
  */
 public final class ResourceManagerUtils {
-    private ResourceManagerUtils() {
+    /**
+     * Creates a new instance of ResourceManagerUtils.
+     *
+     * @deprecated ResourceManagerUtils is a final class.
+     */
+    @Deprecated
+    public ResourceManagerUtils() {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java:35]: private, not `@Deprecated`

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java [hunk 7]
```diff
@@ -249,6 +255,15 @@ public static class InternalRuntimeContext {
         private static DelayProvider delayProvider = new ResourceDelayProvider();
         private static Scheduler reactorScheduler = Schedulers.parallel();
 
+        /**
+         * Creates a new instance of InternalRuntimeContext.
+         *
+         * @deprecated InternalRuntimeContext is a static inner class.
+         */
+        @Deprecated
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java:263]: Not Deprecated

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java]: Fixed in the [commint#7642a91](https://github.com/Azure/azure-sdk-for-java/pull/43192/commits/7642a91a0622459f71ea4750e77e3de93c23746b)

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java (hunk 6 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java]: private

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java]: Fixed in the c[ommit#7aef3c2](https://github.com/Azure/azure-sdk-for-java/pull/43192/commits/7aef3c200b307b9e558e92b532e228d1141a193e)

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/ResourceManager.java [hunk 8]
```diff
@@ -442,6 +478,8 @@ public TagOperations tagOperations() {
     }
 
     /**
+     * Gets the ResourceManagerUtils.InternalRuntimeContext associated with this manager
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/ResourceManager.java:481]: `.` at the end.

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/arm/CountryPhoneCode.java [hunk 9]
```diff
@@ -969,6 +978,8 @@ public static CountryPhoneCode fromString(String code) {
     }
 
     /**
+     * Gets know CountryPhoneCode values.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/arm/CountryPhoneCode.java:981]: typo, `known`

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/models/ForceDeletionResourceType.java [hunk 10]
```diff
@@ -29,6 +38,8 @@ public static ForceDeletionResourceType fromString(String name) {
     }
 
     /**
+     * Gets know ForceDeletionResourceType values.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/models/ForceDeletionResourceType.java:41]: type, `known`

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/models/WhatIfOperationResult.java [hunk 11]
```diff
@@ -17,16 +17,22 @@
...
 
     /**
+     * Gets list of resource changes predicted by What-If operation.
+     *
      * @return list of resource changes predicted by What-If operation.
      */
     List<WhatIfChange> changes();
 
     /**
+     * Gets error when What-If operation fails.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/models/WhatIfOperationResult.java:34]:
```suggestion
     * Gets the error when What-If operation fails.
```

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/models/WhatIfOperationResult.java]: Fixed in the [commit#c526df8](https://github.com/Azure/azure-sdk-for-java/pull/43192/commits/c526df8f1d4631f24b05aada35920868ce4d1535)

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**